src/rust/cryptography-x509-validation/src/trust_store.rs

```diff
@@ -0,0 +1,123 @@
+// This file is dual licensed under the terms of the Apache License, Version
+// 2.0, and the BSD License. See the LICENSE file in the root of this repository
+// for complete details.
+
+use std::vec;
+
+use crate::{
+    certificate::{CertificateError, CertificateExt},
+    extensions::{BasicConstraints, ExtensionError, ExtensionExt, KeyUsage},
+};
+use cryptography_x509::certificate::Certificate;
+
+/// Errors that can occur while constructing a `Store`.
+pub enum StoreError {
+    /// The store would be constructed in an empty state.
+    Empty,
+
+    /// The certificate is well-formed, but isn't acceptable in a store.
+    InvalidMember(&'static str),
```

Comment:
FFR: Picked `&'static str` here to avoid enumerating each error case until they're more stable; happy to break each of the variants out below if you'd prefer.